libavcodec/hevcpred_template.c

```diff
@@ -30,6 +30,15 @@
 static av_always_inline void FUNC(intra_pred)(HEVCContext *s, int x0, int y0,
                                               int log2_size, int c_idx)
 {
```

Comment:
this code wasn't ready for parallel decoding... I'll try to explain briefly:

- this code does HEVC "intra prediction" meaning it figures out the values for pixels from others in the frame.
- it has a concept of whether neighbouring coding blocks are available - cand_bottom_left... etc.
- it interpolates values for neighbouring coding blocks if they're absent - so far so good.
- however, even if they're not available, it still goes and checks if the neighbouring pixels are to be figured out through intra-frame prediction or inter-frame prediction.  This is where it was going wrong - we don't know that yet if the thread responsible for the neighbouring tile hasn't decoded the block yet.  So there were UTs that would fail intermittently.
- luckily, we can get around this by assuming unavailable coding blocks are not intra-predicted.  This is because we're about to reject their pixel values if they're not intra predicted, just like we were if they were unavailable.  So it works to in all cases that we're checking if a pixel was intra predicted, behave the same for non-intra as unavailable.